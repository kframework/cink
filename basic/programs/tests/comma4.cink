int main() {
int x = 5, y = 7;
cout << "x=" << x << ", y=" << y << endl; //x=5, y=7
(x = x + 1,y) = 12;
cout << "x=" << x << ", y=" << y << endl; //x=6, y=12
}
