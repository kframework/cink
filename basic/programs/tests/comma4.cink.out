x=5, y=7
x=6, y=12
<T>
    <in>
        ListItem(#buffer ( "" ))
        ListItem(#istream ( 0 ))
    </in>
    <out>
        ListItem(#ostream ( 1 ))
        ListItem(#buffer ( "" ))
    </out>
    <control>
        <k>
            .K
        </k>
        <return>
            noVal
        </return>
        <fstack>
            .List
        </fstack>
        <sideEffects>
            .Bag
        </sideEffects>
        <env>
            main |-> #symScalLoc(1)
        </env>
    </control>
    <store>
        #symScalLoc(1) |-> lambda (  @ int x ; (x = 5) ; int y ; (y = 7) ;
           (((((cout << "x=") << x) << ", y=") << y) << (endl)) ; (((x = (x + 
          1)) , y) = 12) ; (((((cout << "x=") << x) << ", y=") << y) << (endl)
          ) ; )
        #symScalLoc(2) |-> 6
        #symScalLoc(3) |-> 12
    </store>
    <genv>
        main |-> #symScalLoc(1)
    </genv>
</T>
